File: src/lightnap-ng/src/app/core/backend-api/static-content-status.ts
import { ListItem } from "@core/models/list-item";
import { StaticContentStatus } from "./models";

export const StaticContentStatusListItems = [
  new ListItem<StaticContentStatus>(StaticContentStatus.Draft, "Draft", "Content is not visible to end users."),
  new ListItem<StaticContentStatus>(StaticContentStatus.Published, "Published", "Published content visible to allowed users."),
  new ListItem<StaticContentStatus>(StaticContentStatus.Archived, "Archived", "Archived and not visible to end users."),
];

File: src/LightNap.Core/StaticContents/Enums/StaticContentStatus.cs
namespace LightNap.Core.StaticContents.Enums
{
    /// <summary>
    /// Represents the publication status of static content.
    /// </summary>
    public enum StaticContentStatus
    {
        /// <summary>
        /// Content is in draft state and not yet published.
        /// </summary>
        Draft = 0,

        /// <summary>
        /// Content is published and visible to users.
        /// </summary>
        Published = 1,

        /// <summary>
        /// Content is archived and no longer active.
        /// </summary>
        Archived = 2,
    }
}
